### PR TITLE
Add Language Igor

### PR DESCRIPTION
**Pull request by ukos-git**
This adds support for Igor. Igor is a scientific language developed by
[WaveMetrics](https://www.wavemetrics.com/) and used in their Software `Igor Pro` with a
[few projects on Github](https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Aipf+rtGlobals+NOT+nothack)

This lexer was translated directly from the [pygments
lexer](http://pygments.org/docs/lexers/#lexers-for-igor-pro) without any
changes.

### CODE REVIEW

**Comment by ukos-git** [comment]
The pygments lexer is very basic but working for most of the cases properly so it is ok to use it.

File: lexers/testdata/igor.expected [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+  {
+    "type": "NameDecorator",
+    "value": "#pragma"
+  },
+  {
+    "type": "Text",
+    "value": " rtGlobals=3\n"
+  },
+  {
+    "type": "Error",
```
Comment (alecthomas) [inline lexers/testdata/igor.expected]: The error tokens indicate that the lexer isn't fully matching.

Comment (ukos-git) [reply lexers/testdata/igor.expected]: thanks. The line endings are not matching.

**Comment by alecthomas** [comment]
Thanks!

**Pull request merged**